crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -752,22 +752,41 @@ impl PublishFlakeRef {
         flox: &Flox,
         git_service_prefer_https: bool,
     ) -> Result<Self, ConvertFlakeRefError> {
-        let publish_flake_ref = match flake_ref {
-            FlakeRef::GitSsh(ssh_ref) => Self::Ssh(ssh_ref),
-            FlakeRef::GitHttps(https_ref) => Self::Https(https_ref),
+        // This should really be a recursive function, but that requires two things:
+        // - Making this return type Pin<Box<impl Future<Output = ...>>>
+        // - That all Futures that this one depends on are Send
+        // It turns out we can't do that because the GitProvider trait doesn't require the Futures it
+        // returns to be Send. Instead of a recursive call we do this loop until we no longer have
+        // an indirect flake reference. It should run a maximum of twice (once if `flake_ref` isn't indirect,
+        // twice if it is indirect).
+        let flake_ref = if let FlakeRef::Indirect(indirect) = flake_ref {
+            indirect.resolve()? 
+        } else {
+            flake_ref
+        };
+        let publish_flakeref = match flake_ref.clone() {
+            FlakeRef::GitSsh(ssh_ref) => {
+                Self::Ssh(ssh_ref);
+            },
+            FlakeRef::GitHttps(https_ref) => {
+                Self::Https(https_ref);
+            },
             // resolve upstream for local git repo
             FlakeRef::GitPath(file_ref) => {
-                Self::from_git_file_flake_ref(file_ref, &flox.nix(Default::default())).await?
+                Self::from_git_file_flake_ref(file_ref, &flox.nix(Default::default()))
+                    .await?;
             },
-            // resolve indirect ref to direct ref (recursively)
-            FlakeRef::Indirect(_) => todo!(),
             FlakeRef::Github(github_ref) => {
-                Self::from_github_ref(github_ref, git_service_prefer_https)?
+                break Self::from_github_ref(github_ref, git_service_prefer_https)?;
```

Comment:
```suggestion
                Self::from_github_ref(github_ref, git_service_prefer_https)?;
```

Reply:
sorry, i was making these changes on the web version, there might be some subtle typing issues with it